pkg/operation/botanist/component/nginxingressshoot/nginxingress.go

```diff
@@ -0,0 +1,699 @@
+				RevisionHistoryLimit: pointer.Int32(1),
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						v1beta1constants.LabelApp: labelAppValue,
+						labelKeyComponent:         labelValueBackend,
+						labelKeyRelease:           labelValueAddons,
+					},
+				},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{
+							v1beta1constants.LabelApp:   labelAppValue,
+							labelKeyComponent:           labelValueBackend,
+							v1beta1constants.GardenRole: v1beta1constants.GardenRoleOptionalAddon,
+							labelKeyRelease:             labelValueAddons,
+							"origin":                    "gardener",
+						},
+					},
+					Spec: corev1.PodSpec{
+						PriorityClassName:             "system-cluster-critical",
+						NodeSelector:                  map[string]string{v1beta1constants.LabelWorkerPoolSystemComponents: "true"},
+						TerminationGracePeriodSeconds: pointer.Int64(60),
+						SecurityContext: &corev1.PodSecurityContext{
+							RunAsUser:          pointer.Int64(65534),
+							FSGroup:            pointer.Int64(65534),
+							SupplementalGroups: []int64{1},
+							SeccompProfile: &corev1.SeccompProfile{
+								Type: corev1.SeccompProfileTypeRuntimeDefault,
+							},
+						},
+						Containers: []corev1.Container{{
+							Name:            containerNameBackend,
+							Image:           n.values.DefaultBackendImage,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							LivenessProbe: &corev1.Probe{
+								ProbeHandler: corev1.ProbeHandler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path:   "/healthy",
+										Port:   intstr.FromInt(int(containerPortBackend)),
+										Scheme: corev1.URISchemeHTTP,
+									},
+								},
+								InitialDelaySeconds: 30,
+								TimeoutSeconds:      5,
+							},
+							Ports: []corev1.ContainerPort{{
+								ContainerPort: containerPortBackend,
+								Protocol:      corev1.ProtocolTCP,
+							}},
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("20m"),
+									corev1.ResourceMemory: resource.MustParse("20Mi"),
+								},
+								Limits: corev1.ResourceList{
+									corev1.ResourceMemory: resource.MustParse("100Mi"),
+								},
+							},
+						}},
+					},
+				},
+			},
+		}
+
+		deploymentController = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      deploymentNameController,
+				Namespace: metav1.NamespaceSystem,
+				Labels: map[string]string{
+					v1beta1constants.GardenRole: v1beta1constants.GardenRoleOptionalAddon,
+					v1beta1constants.LabelApp:   labelAppValue,
+					labelKeyRelease:             labelValueAddons,
+					labelKeyComponent:           labelValueController,
+					"origin":                    "gardener",
+				},
+			},
+			Spec: appsv1.DeploymentSpec{
+				Replicas:             pointer.Int32(1),
+				RevisionHistoryLimit: pointer.Int32(1),
```

Comment:
```suggestion
				RevisionHistoryLimit: pointer.Int32(2),
```

Nit: We choose two for most components